src/main/resources/custom-ui.js

```diff
@@ -1,12 +1,109 @@
+if(!$('#jqueryInputMaskJs').length) {
+    var s = document.createElement("script");
+    s.id = "jqueryInputMaskJs";
+    s.type = "text/javascript";
+    s.src = "../prevac/resources/js/jquery.inputmask.js";
+    $("head").append(s);
+}
+
+if(!$('#inputMaskJs').length) {
+    var s = document.createElement("script");
+    s.id = "inputMaskJs";
+    s.type = "text/javascript";
+    s.src = "../prevac/resources/js/inputmask.js";
+    $("head").append(s);
+}
+
 $scope.showBackToEntityListButton = false;
 $scope.showAddInstanceButton = false;
-$scope.showLookupButton = true;
-$scope.showFieldsButton = false;
-$scope.showImportButton = false;
-$scope.showExportButton = false;
-$scope.showViewTrashButton = false;
-$scope.showFiltersButton = false;
 $scope.showDeleteInstanceButton = false;
+$scope.showLookupButton = true;
+
+if ($scope.selectedEntity.name === "Participant") {
+    $rootScope.selectedTab = "subjects";
+} else {
+    $rootScope.selectedTab = "visitLimitation";
+    $scope.showFieldsButton = false;
+    $scope.showImportButton = false;
+    $scope.showExportButton = false;
+    $scope.showViewTrashButton = false;
+    $scope.showFiltersButton = false;
+}
+
+var importCsvModal = '../prevac/resources/partials/modals/import-csv.html';
+var editSubjectModal = '../prevac/resources/partials/modals/edit-subject.html';
+
+$scope.customModals.push(importCsvModal);
+$scope.customModals.push(editSubjectModal);
+
+$scope.importEntityInstances = function() {
+    $('#importSubjectModal').modal('show');
+};
+
+$scope.importSubject = function () {
+    blockUI();
+
+    $('#importSubjectForm').ajaxSubmit({
+        success: function () {
+            $("#instancesTable").trigger('reloadGrid');
+            $('#importSubjectForm').resetForm();
+            $('#importSubjectModal').modal('hide');
+            unblockUI();
```

Comment:
in case of an error the message is displayed and the modal is not dismissed, so you can choose another file (if modal is still visible UI should not be unblocked)

Reply:
Okay, I see. That is correct.